src/compiler/resolutionCache.ts

```diff
@@ -388,16 +389,19 @@ namespace ts {
                 !redirectedReference || redirectedReference.sourceFile.path !== oldRedirect.sourceFile.path :
                 !!redirectedReference;
 
-            const seenNamesInFile = new Map<string, true>();
+            const seenNamesInFile = createModeAwareCache<true>();
+            let i = 0;
             for (const name of names) {
-                let resolution = resolutionsInFile.get(name);
+                const mode = containingSourceFile ? getModeForResolutionAtIndex(containingSourceFile, i) : undefined;

```

Comment:
We cannot assume that module name is referenced at index `i` in containing source file because we could be calling you with only few module names from the source file. https://github.com/microsoft/TypeScript/blob/main/src/compiler/program.ts#L1571 shows that case